### PR TITLE
feat(adaptor): Support AWS Function URL Streaming

### PR DESCRIPTION
**Pull request by watany-dev**
fixed: https://github.com/honojs/hono/issues/1183

We were able to run the streaming response by operating this module on the actual machine. Unfortunately, awslambda.streamifyResponse provided by AWS Lambda is globally available in the Lambda Nodejs environment, but no method for local testing is provided.

Therefore, we decided to separate it as streamHandle from the existing handle. This way, even if there's a bug, existing users won't be affected. Whether to release this as 'experimental' or to halt integration until a testing method is established is a challenging decision.

### Author should do the followings, if applicable

- [ ] Add tests
- [ ] Run tests
- [ ] `yarn denoify` to generate files for Deno

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @watany-dev,

> Unfortunately, `awslambda.streamifyResponse` provided by AWS Lambda is globally available in the Lambda Node.js environment, but no method for local testing is provided.

Can we create a fake API for mocking in testing only? For example, the `Cache API` does not exist on Node.js, but we mock it in [this file](https://github.com/honojs/hono/blob/main/src/test-utils/setup-vitest.ts) to test Cache Middleware.

```ts
type StoreMap = Map<string | Request, Response>

class MockCache {
  // ...
}

const globalStore: Map<string | Request, Response> = new Map()

const caches = {
  open: (name: string) => {
    return new MockCache(name, globalStore)
  },
}

vi.stubGlobal('caches', caches)
```

I think it's OK to mock that are difficult to reproduce depending on the specific environment.

**Comment by watany-dev** [comment]
@yusukebe 
I've taken into account your helpful advice, and I'm pleased to report that the local tests have passed without issue. Thank you.

**Comment by yusukebe** [comment]
@watany-dev 

Great! Looks good. I'll merge this later.

File: .pre-commit-config.yaml
Comment (yusukebe) [inline .pre-commit-config.yaml:1]: You forgot remove this?

**Review by yusukebe — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml
Comment (yusukebe) [reply .pre-commit-config.yaml:1]: Ah, sorry. Already removed.

**Review by yusukebe — APPROVED** [review]

**Comment by yusukebe** [comment]
It's a time for merging! I'll merge this into the "next" branch and will release a RC version first. Thanks!

**Pull request merged**